packages/react-app/src/modules/Counter/counter.middleware.js

```diff
@@ -0,0 +1,49 @@
+import { prop, includes, path } from 'ramda'
+
+import { getCounterIdFromPayload } from './counter.utils'
+import {
+  NAVIGATE_TO_COUNTERS, NAVIGATE_TO_SELECTED_COUNTER, selectCounter,
+} from './counters.reducer'
+
+//---------------------------------
+// navigate to single counter
+//---------------------------------
+
+export const navigateToCounterFlow = navigate => ({ dispatch }) => next => (action) => {
+  next(action)
+
+  const { type, payload } = action
+  if (type === NAVIGATE_TO_SELECTED_COUNTER ) {
```

Comment:
this should just be SELECT_COUNTER
and also I think middleware wherever possible should use state instead of payload,
so here get id from state.
eg.
```
    const { selectedCounter } = getState()
    const newUrl = `/counter/${selectedCounter}`
```
that way if reducer logic is the on place where selectedCounter is determined